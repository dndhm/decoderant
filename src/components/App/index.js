import React, { Fragment } from 'react';
import propTypes from 'prop-types';
import injectStaticSheet from 'inject-static-sheet';

import Heading from '../Heading';
import Grid from '../Grid';
import Project from '../Project';

const styles = {
	heading: {
		padding: '40px 0',
	},
};

const App = ({
	classes,
}) => (
	<Fragment>
		<div className={classes.heading}>
			<Heading level={1}>Decoderant</Heading>
		</div>
		<Grid>
			<Project
				description="In 2016/17 I worked at The Financial Times in their new, but rapidly evolving Internal Tooling department. I lead UI development on a hub for managing GDPR Subject Access Requests and quickly built a robust and scalable application using React, Redux and NodeJS."
				tags={['NodeJS', 'React', 'Redux', 'Neo4j', 'CSS']}
				title="FT Internal Tooling"
			/>
			<Project
				description="As part of a multi-billion pound transformation project, the Account Opening project was at the forefront of a massive shift in HSBC's online offering. I was one of two developers at the outset of the project back in 2015 and saw the project right through to its launch in 2017."
				tags={['React', 'Redux', 'Responsive', 'CSS']}
				title="HSBC Customer Onboarding"
			/>
			<Project
				description="Working with Ostmodern I developed this simple, yet rich site for Cirkus, a new on-demand video service."
				tags={['CSS', 'JS', 'Responsive']}
				title="Cirkusgo.com"
			/>
			<Project
				description="I worked as an integral part of the BallBall team at Ostmodern, building the Backbone Marionette app architecture and much of the user interface for this slick Android app."
				tags={['Backbone', 'JS', 'Phonegap']}
				title="BallBall Android App"
			/>
			<Project
				description="I built the front end for Ostmodern's beautiful new website. Its minimal design belies a suprisingly rich user experience."
				tags={['CSS', 'JS', 'Responsive']}
				title="Ostmodern"
			/>
			<Project
				description="This parallax scrolling site makes use of striking photography and bold typography to draw the user through the agency's key values. It is content managed via a heavily customised, Wordpress-based CMS."
				tags={['CSS', 'JS', 'Wordpress', 'PHP']}
				title="Grace Blue"
			/>
			<Project
				description="Working with Ostmodern, I was responsible for the front end development of EE Film. The site features deceptively powerful film search tools and some subtle but effective usability features."
				tags={['CSS', 'JS']}
				title="EE Film"
			/>
			<Project
				description="Again working with Ostmodern, I added engaging front-end features and polish to YouView.com in preparation for the new VOD service&rsquo;s high-profile 2012 launch."
				tags={['CSS', 'JS']}
				title="YouView"
			/>
			<Project
				description="1DA is a new creative consultancy based in London who tasked me with creating a minimal, yet impactful site that would allow their content to speak for its self."
				tags={['CSS', 'JS', 'Wordpress', 'PHP']}
				title="1DA"
			/>
			<Project
				description="This site provides information about Merchant Square and its surrounding area for prospective buyers. Content and users are managed by the client using an intuitive interface."
				tags={['CSS', 'JS', 'Wordpress', 'PHP']}
				title="3 Merchant Square"
			/>
			<Project
				description="Launched in 2011, the Bostock and Pollitt website represents the agency&rsquo;s approach through the use of full-bleed imagery, striking visuals and tactile navigational devices."
				tags={['CSS', 'JS', 'Wordpress', 'PHP']}
				title="Bostock and Pollitt"
			/>
			<Project
				description="Londonwallplace.com uses a variety of techniques to make its subject matter stand out, including a custom Google Map driven by 3rd party APIs and even floorplans that are more fun to use than they have any right to be."
				tags={['CSS', 'JS', 'Wordpress', 'PHP']}
				title="London Wall Place"
			/>
			<Project
				description="Rather than simply presenting a static online version of their printed report, I injected movement and texture into Morgan Sindall&rsquo;s 2010 annual report. The site is peppered with animated charts and tactile navigation devices to help the user navigate its burgeoning content."
				tags={['CSS', 'JS', 'Wordpress', 'PHP']}
				title="Morgan Sindall AR 2010"
			/>
			<Project
				description="World&rsquo;s Biggest Coffee Morning is Macmillan Cancer Support&rsquo;s flagship fundraiser. I built the front-end templates for their team to implement, including a data-driven coffee bean counter."
				tags={['CSS', 'JS']}
				title="World's Biggest Coffee Morning"
			/>
			<Project
				description="Codeofconviction.com formed the hub of a Gold DMA award-winning campaign for Ubisoft that spanned Twitter, fake security company websites and YouTube."
				tags={['CSS', 'JS', 'Social media', 'PHP']}
				title="Code of Conviction"
			/>
		</Grid>
	</Fragment>
);

App.propTypes = {
	classes: propTypes.object,
};

export default injectStaticSheet(styles, 'App')(App);
