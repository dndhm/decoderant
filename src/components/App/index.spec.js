import React from 'react';
import { create } from 'react-test-renderer';

import App from './';

describe('App', () => {
	test('should render correctly', () => {
		const tree = create(<App />).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
