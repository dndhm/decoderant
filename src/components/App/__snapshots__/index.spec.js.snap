// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
Array [
  <div
    className="App--heading"
>
    <h1
        className="Heading--heading Heading--h1 Heading--color-black"
    >
        Decoderant
    </h1>
</div>,
  <div
    className="Grid--container"
>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                FT Internal Tooling
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    NodeJS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    React
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Redux
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Neo4j
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    In 2016/17 I worked at The Financial Times in their new, but rapidly evolving Internal Tooling department. I lead UI development on a hub for managing GDPR Subject Access Requests and quickly built a robust and scalable application using React, Redux and NodeJS.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                HSBC Customer Onboarding
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    React
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Redux
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Responsive
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    As part of a multi-billion pound transformation project, the Account Opening project was at the forefront of a massive shift in HSBC's online offering. I was one of two developers at the outset of the project back in 2015 and saw the project right through to its launch in 2017.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                Cirkusgo.com
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    JS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Responsive
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    Working with Ostmodern I developed this simple, yet rich site for Cirkus, a new on-demand video service.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                BallBall Android App
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    Backbone
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    JS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Phonegap
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    I worked as an integral part of the BallBall team at Ostmodern, building the Backbone Marionette app architecture and much of the user interface for this slick Android app.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                Ostmodern
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    JS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Responsive
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    I built the front end for Ostmodern's beautiful new website. Its minimal design belies a suprisingly rich user experience.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                Grace Blue
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    JS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Wordpress
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    PHP
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    This parallax scrolling site makes use of striking photography and bold typography to draw the user through the agency's key values. It is content managed via a heavily customised, Wordpress-based CMS.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                EE Film
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    JS
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    Working with Ostmodern, I was responsible for the front end development of EE Film. The site features deceptively powerful film search tools and some subtle but effective usability features.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                YouView
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    JS
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    Again working with Ostmodern, I added engaging front-end features and polish to YouView.com in preparation for the new VOD service’s high-profile 2012 launch.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                1DA
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    JS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Wordpress
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    PHP
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    1DA is a new creative consultancy based in London who tasked me with creating a minimal, yet impactful site that would allow their content to speak for its self.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                3 Merchant Square
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    JS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Wordpress
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    PHP
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    This site provides information about Merchant Square and its surrounding area for prospective buyers. Content and users are managed by the client using an intuitive interface.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                Bostock and Pollitt
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    JS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Wordpress
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    PHP
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    Launched in 2011, the Bostock and Pollitt website represents the agency’s approach through the use of full-bleed imagery, striking visuals and tactile navigational devices.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                London Wall Place
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    JS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Wordpress
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    PHP
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    Londonwallplace.com uses a variety of techniques to make its subject matter stand out, including a custom Google Map driven by 3rd party APIs and even floorplans that are more fun to use than they have any right to be.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                Morgan Sindall AR 2010
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    JS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Wordpress
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    PHP
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    Rather than simply presenting a static online version of their printed report, I injected movement and texture into Morgan Sindall’s 2010 annual report. The site is peppered with animated charts and tactile navigation devices to help the user navigate its burgeoning content.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                World's Biggest Coffee Morning
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    JS
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    World’s Biggest Coffee Morning is Macmillan Cancer Support’s flagship fundraiser. I built the front-end templates for their team to implement, including a data-driven coffee bean counter.
                </p>
            </div>
        </div>
    </div>
    <div
        className="Project--container"
    >
        <div
            className="Project--content"
        >
            <h3
                className="Heading--heading Heading--h3 Heading--color-white"
            >
                Code of Conviction
            </h3>
            <ul
                className="ProjectTags--list"
            >
                <li
                    className="ProjectTags--listItem"
                >
                    CSS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    JS
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    Social media
                </li>
                <li
                    className="ProjectTags--listItem"
                >
                    PHP
                </li>
            </ul>
            <div
                className="Project--description"
            >
                <p
                    className="Paragraph--paragraph"
                >
                    Codeofconviction.com formed the hub of a Gold DMA award-winning campaign for Ubisoft that spanned Twitter, fake security company websites and YouTube.
                </p>
            </div>
        </div>
    </div>
</div>,
]
`;
